src/m365/teams/commands/user/user-app-remove.ts

```diff
@@ -95,14 +112,30 @@ class TeamsUserAppRemoveCommand extends GraphCommand {
         type: 'confirm',
         name: 'continue',
         default: false,
-        message: `Are you sure you want to remove the app with id ${args.options.id} for user ${args.options.userId}?`
+        message: `Are you sure you want to remove the app with id ${args.options.id} for user ${args.options.userId ?? args.options.userName}?`
       });
 
       if (result.continue) {
         await removeApp();
       }
     }
   }
+
+  private async getUserId(args: CommandArgs): Promise<{ value: string }> {
```

Comment:
We don't need to resolve the UPN to a user ID. Seems like passing the UPN works just fine: https://developer.microsoft.com/en-us/graph/graph-explorer?request=users%2FMeganB%40M365x214355.onmicrosoft.com%2Fteamwork%2FinstalledApps&method=GET&version=v1.0&GraphUrl=https://graph.microsoft.com

Reply:
@waldekmastykarz : Thanks for the suggestion , much appreciated , always great to know better alternatives. I have removed the function getUserId.